openedx/core/djangoapps/config_model_utils/models.py

```diff
@@ -0,0 +1,125 @@
+"""
+Useful ConfigurationModel subclasses
+
+StackedConfigurationModel: A ConfigurationModel that can be overridden at site, org and course levels
+"""
+
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from collections import namedtuple
+
+from django.conf import settings
+from django.db import models
+from django.contrib.sites.models import Site
+from django.utils.translation import ugettext_lazy as _
+
+from config_models.models import ConfigurationModel
+from openedx.core.djangoapps.site_configuration.models import SiteConfiguration
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+
+class StackedConfigurationModel(ConfigurationModel):
+    """
+    A ConfigurationModel that stacks Global, Site, Org, and Course level
+    configuration values.
+    """
+    class Meta(object):
+        abstract = True
+
+    KEY_FIELDS = ('site', 'org', 'course')
+
+    enabled = models.NullBooleanField(default=None, verbose_name=_("Enabled"))
+    site = models.ForeignKey(Site, on_delete=models.CASCADE, null=True)
+    org = models.CharField(max_length=255, db_index=True, null=True)
+    course = models.ForeignKey(
+        CourseOverview,
+        on_delete=models.DO_NOTHING,
+        null=True,
+    )
+
+    @classmethod
+    def attribute_tuple(cls):
+        "Returns a namedtuple with all attributes that can be overridden on this config model."
+        if hasattr(cls, '_attribute_tuple'):
+            return cls._attribute_tuple
+
+        cls._attribute_tuple = namedtuple(
+            '{}Values'.format(cls.__name__),
+            cls.STACKABLE_FIELDS
+        )
+        return cls._attribute_tuple
+
+    @classmethod
+    def current(cls, site=None, org=None, course=None):  # disable: pylint=arguments-differ
+        """
+        Return the current overridden configuration at the specified level.
+
+        Only one level may be specified at a time. Specifying multiple levels
+        will result in a ValueError.
+
+        Arguments:
+            site: The Site to check current values for
+            org: The org to check current values for
+            course: The course to check current values for
+
+        Returns:
+            An instance of :class:`cls.attribute_tuple()` with the overridden values
+            specified down to the level of the supplied argument (or global values if
+            no arguments are supplied).
```

Comment:
imo adding an example call with an example response to the docstring would make this function easier to understand

Reply:
Yup, agreed. I'll add it.